mlflow/tracing/assessment.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any, Optional, Union
+
+from mlflow.entities.assessment import (
+    Assessment,
+    AssessmentError,
+    AssessmentValueType,
+    Expectation,
+    Feedback,
+    experimental,
+)
+from mlflow.entities.assessment_source import AssessmentSource
+from mlflow.exceptions import MlflowException
+from mlflow.tracking.client import MlflowClient
+
+
+@experimental
+def log_expectation(
+    trace_id: str,
+    name: str,
+    value: AssessmentValueType,
+    source: Union[str, AssessmentSource],
+    metadata: Optional[dict[str, Any]] = None,
+    span_id: Optional[str] = None,
+) -> Assessment:
+    """
+
+    .. important::
+
+        This API is currently only available for [Databricks Managed MLflow](https://www.databricks.com/product/managed-mlflow).
+
+    Logs an expectation (ground truth) to a Trace.
+
+
+    Args:
+        trace_id: The ID of the trace.
+        name: The name of the expectation assessment e.g., "expected_answer
+        value: The value of the expectation. It can be any JSON-serializable value.
+        source: The source of the expectation assessment. Must be either an instance of
+                :py:class:`~mlflow.entities.AssessmentSource` or a string that
+                is a valid value in the
+                :py:class:`~mlflow.entities.AssessmentSourceType` enum.
+        metadata: Additional metadata for the expectation.
+        span_id: The ID of the span associated with the expectation, if it needs be
+                associated with a specific span in the trace.
+
+    Returns:
+        :py:class:`~mlflow.entities.Assessment`: The created expectation assessment.
+
+    Example:
+
+    The following code annotates a trace with human-provided ground truth.
+
+    .. code-block:: python
+
+        import mlflow
+        from mlflow.entities.assessment import AssessmentSourceType
+
+        mlflow.log_expectation(
+            trace_id="1234",
+            name="expected_answer",
+            value=42,
+            source=AssessmentSourceType.HUMAN,
+        )
+
+    """
+    if value is None:
+        raise MlflowException.invalid_parameter_value("Expectation value cannot be None.")
+
+    return MlflowClient().log_assessment(
+        trace_id=trace_id,
+        name=name,
+        source=_parse_source(source),
+        value=Expectation(value) if value is not None else None,
+        metadata=metadata,
+        span_id=span_id,
+    )
+
+
+@experimental
+def log_feedback(
+    trace_id: str,
+    name: str,
+    source: Union[str, AssessmentSource],
+    value: Optional[AssessmentValueType] = None,
+    error: Optional[AssessmentError] = None,
+    rationale: Optional[str] = None,
+    metadata: Optional[dict[str, Any]] = None,
+    span_id: Optional[str] = None,
```

Comment:
nit: same comment as above about keeping arg order as similar as possible

```suggestion
    metadata: Optional[dict[str, Any]] = None,
    span_id: Optional[str] = None,
    error: Optional[AssessmentError] = None,
    rationale: Optional[str] = None,
```

Reply:
Hmm logically `rational` closely ties to `value` and `error` has some condition between `value`, so I'd like to keep the current order. Since `log_feedback` and `log_expectation` are typically used in different step of the development cycle, consistency between two is less important than making args order logically makes sense?